server/datastore/s3/installer.go

```diff
@@ -0,0 +1,62 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+const (
+	desktopPath = "desktop"
+	executable  = "fleet-osquery"
+)
+
+// InstallerStore contains methods to retrieve installers from S3
+type InstallerStore struct {
+	*s3store
+}
+
+// NewInstallerStore creates a new instance with the given S3 config
+func NewInstallerStore(config config.S3Config) (*InstallerStore, error) {
+	s3store, err := newS3store(config)
+	if err != nil {
+		return nil, err
+	}
+	return &InstallerStore{s3store}, nil
+}
+
+// Exists checks if an installer exists in the S3 bucket
+func (i *InstallerStore) Exists(ctx context.Context, installer fleet.Installer) (bool, error) {
+	key := i.keyForInstaller(installer)
+	_, err := i.s3client.HeadObject(&s3.HeadObjectInput{Bucket: &i.bucket, Key: &key})
```

Comment:
Maybe we could use something like https://stackoverflow.com/questions/57697095/how-i-can-safely-check-if-file-exists-in-s3-bucket-using-go-in-lambda?

Reply:
I have something similar to this in the branch that actually uses this method (check `origin/6365-s3-api`, it's using `ErrCodeNoSuchKey`, but according to that link it doesn't actually work with S3, so will add a safety check)

Would you be OK merging this as-is knowing this? The improved error handling makes more sense in the context of that branch too